src/app/regular-giving/regular-giving.component.ts

```diff
@@ -39,16 +41,25 @@ export class RegularGivingComponent implements OnInit {
   @ViewChild('stepper') private stepper: MatStepper;
   readonly termsUrl = 'https://biggive.org/terms-and-conditions';
   readonly privacyUrl = 'https://biggive.org/privacy';
+  protected donor: Person;
 
   constructor(
     private route: ActivatedRoute,
     private imageService: ImageService,
     private formBuilder: FormBuilder,
     private snackBar: MatSnackBar,
+    private identityService: IdentityService,
+    private router: Router,
   ) {
   }
 
   ngOnInit() {
+    const donor: Person | null = this.route.snapshot.data.donor;
+    if (! donor) {
+      throw new Error("Must be logged in to see regular giving page");
```

Comment:
What will happen when SSR mode gets a request for this?

Reply:
This page requires login which SSR doesn't do, so navigation will be cancelled at this line: https://github.com/thebiggive/donate-frontend/blob/d92d7b2fa70cce9471759794aed19cdf99397333/src/app/app-routing.ts#L44